rspace/src/main/scala/coop/rchain/rspace/history/RadixTree.scala

```diff
@@ -0,0 +1,999 @@
+              // Fill first byte - item number
+              arr(pos0) = numItem.toByte
+              // Fill second byte - NodePtr identifier (most significant bit = 1) and prefixSize (lower 7 bits = size)
+              val posSecondByte   = pos0 + 1
+              val prefixSize: Int = prefix.size.toInt
+              arr(posSecondByte) = encodeNodePtrSecondByte(prefixSize)
+              // Fill prefix
+              val posPrefixStart = posSecondByte + 1
+              for (i <- 0 until prefixSize) arr(posPrefixStart + i) = prefix(i.toLong)
+              // Fill ptr
+              val posPtrStart = posPrefixStart + prefixSize
+              for (i <- 0 until defSize) arr(posPtrStart + i) = ptr(i.toLong)
+              putItemIntoArray(numItem + 1, posPtrStart + defSize) // Loop to the next item.
+          }
+        }
+      ByteVector(putItemIntoArray(0, 0))
+    }
+
+    /** Deserialization [[ByteVector]] to [[Node]]
+      */
+    def decode(bv: ByteVector): Node = {
+      val arr     = bv.toArray
+      val maxSize = arr.length
+
+      // If first bit 0 - return true, otherwise false.
+      def isLeaf(secondByte: Byte) = (secondByte & 0x80) == 0x00
+
+      @tailrec
+      // Each loop decodes one non-empty item.
+      def decodeItem(pos0: Int, node: Node): Node =
+        if (pos0 == maxSize) node // End of deserialization
+        else {
+          val (idx0Next, nodeNext) = try {
+            val numItem: Int = byteToInt(arr(pos0)) // Take first byte - it's item's number
+            assert(
+              node(numItem) == EmptyItem,
+              "Error during deserialization: wrong number of item."
+            )
+            val pos1       = pos0 + 1
+            val secondByte = arr(pos1) // Take second byte
+
+            // Decoding prefix
+            val prefixSize: Int = secondByte & 0x7F // Lower 7 bits - it's size of prefix (0..127).
+            val prefix          = new Array[Byte](prefixSize)
+            val posPrefixStart  = pos1 + 1
+            for (i <- 0 until prefixSize) prefix(i) = arr(posPrefixStart + i) // Take prefix
+
+            // Decoding leaf or nodePtr data
+            val valOrPtr         = new Array[Byte](defSize)
+            val posValOrPtrStart = posPrefixStart + prefixSize
+            for (i <- 0 until defSize)
+              valOrPtr(i) = arr(posValOrPtrStart + i) // Take next 32 bytes - it's data
+
+            val pos0Next = posValOrPtrStart + defSize // Calculating start position for next loop
+
+            // Decoding type of non-empty item
+            val item = if (isLeaf(secondByte)) {
+              Leaf(ByteVector(prefix), ByteVector(valOrPtr))
+            } else {
+              NodePtr(ByteVector(prefix), ByteVector(valOrPtr))
+            }
+
+            (pos0Next, node.updated(numItem, item))
+          } catch {
+            case _: Exception =>
+              assert(assertion = false, "Error during deserialization: out of range")
```

Comment:
What is the purpose of this code? It just rethrows the error.

Reply:
If we have bad data in KVDB, or if we have data from another codec format, we get this error. I can change this message. For example: "Error during deserialization: Invalid data format".